js/map.js

```diff
@@ -96,8 +124,101 @@ const resetMarker = () => {
   address.value = `${mainMarker.getLatLng().lat.toFixed(5)}, ${mainMarker.getLatLng().lng.toFixed(5)}`;
 };
 
+// Сброс формы
 resetButton.addEventListener('click', () => {
   resetForm();
 });
 
-export {renderMarkers, resetMarker};
+// СОБЫТИЯ ФИЛЬТРОВ
+// Добавляем события фильтров
+filterTypeElement.addEventListener('change', debounce(getData, DEFAULT_DEBOUNCE));
+filterPriceElement.addEventListener('change', debounce(getData, DEFAULT_DEBOUNCE));
+filterRoomsNumberElement.addEventListener('change', debounce(getData, DEFAULT_DEBOUNCE));
+filterGuestsNumberElement.addEventListener('change', debounce(getData, DEFAULT_DEBOUNCE));
+filterFeaturesElementList.forEach((element) =>
+  element.addEventListener('click', debounce(getData, DEFAULT_DEBOUNCE)));
+
+// ФИЛЬТРЫ
+//Фильтруем по типу жилья
+function filterByType (offers) {
+  if (filterTypeElement.value !== DEFAULT_VALUE) {
+    offers = offers.filter((offer) => (offer.offer.type) ? offer.offer.type === filterTypeElement.value : false);
```

Comment:
Разве тут нельзя просто сразу вернуть из функции `offer.offer.type === filterTypeElement.value`?

И еще момент - тут лучше аргумент из filter переименовать с offer на что-то другое, потому что offer.offer сильно замыливает глаз

То есть в целом ты можешь сделать, я думаю
```suggestion
    offers = offers.filter((offer) => offer.offer.type === filterTypeElement.value);
```

Или еще вариант - если offer.type может не быть - то ты можешь написать так 
```suggestion
    offers = offers.filter((offer) => (offer.offer.type === filterTypeElement.value) || false);
```